presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -663,7 +664,8 @@ public void dropColumn(HiveIdentity identity, String databaseName, String tableN
                             .withCatalogId(catalogId)
                             .withDatabaseName(table.getDatabaseName())
                             .withTableName(table.getTableName())
-                            .withPartitionValues(partitionValues)));
+                            .withPartitionValues(partitionValues)
```

Comment:
this method only fetches single partition, `io.prestosql.plugin.hive.metastore.glue.GlueHiveMetastore#batchGetPartition` is the method you want to change. However, doesn't `com.amazonaws.services.glue.AWSGlueAsync#batchGetPartitionAsync(com.amazonaws.services.glue.model.BatchGetPartitionRequest)` already return all partitions in single request?

Reply:
@sopel39 thanks, I will update it it. The behaviour we see suggests otherwise.

Reply:
> The behaviour we see suggests otherwise.

did you debug code execution? Why `getParition` (non-batch) version is called many times?